grails-app/controllers/streama/UserController.groovy

```diff
@@ -59,10 +61,11 @@ class UserController {
 
   def checkAvailability() {
     def username = params.username
+    def isInvite = params.isInvite
     def result = [:]
 
     if (User.findByUsername(username)) {
-      result.error = "User with that E-Mail-Address already exists."
+      result.error = (isInvite == "true") ? "User with that E-Mail-Address already exists." : "Username already exists."
```

Comment:
I think server shouldn't really care about customized error messages.
You can only pass username and to check against database records. Send *HEAD* or *GET* request and see if user does not exists return *404*, if user is in db return *200*. Based on server response you can show relevant message on client side. Besides flag argument will be removed in `checkAvailability` function 👍